app/src/androidTest/java/com/droidhats/campuscompass/views/NavigationDrawerTest.kt

```diff
@@ -0,0 +1,91 @@
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import com.droidhats.campuscompass.MainActivity
+import com.droidhats.campuscompass.R
+import junit.framework.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class NavigationDrawerTest {
+
+    @get:Rule
+    var activityRule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)
+
+    private lateinit var navController: TestNavHostController
+
+    @Before
+    fun setUp() {
+        navController = TestNavHostController(
+            ApplicationProvider.getApplicationContext())
+        //needs to be here to see which fragment is being opened
+        navController.setGraph(R.navigation.navigation)
+        Thread.sleep(3000)
+    }
+
+    @Test
+    fun test_isNavigationCorrect(){
+
+        //Opening side menu by clicking on menu  button
+        onView(withId(R.id.mt_nav)).check(matches(isDisplayed())).perform(click())
+
+        //checking if side menu is opened
+        onView(withId(R.id.drawer_layout)).check(ViewAssertions.matches(DrawerMatchers.isOpen()));
+
+        //clicking outside of the menu to check if it closes
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation()).click(1200, 200)
+    }
+
+    @Test
+    fun test_isSideMenuListElementsInView(){
+
+        //Opening side menu by clicking on menu  button
+        onView(withId(R.id.mt_nav)).check(matches(isDisplayed())).perform(click())
+
+        //Checking if logo is displayed
+        onView(withId(R.id.nav_logo)).check(matches(isDisplayed()))
+
+        //Checking if title is displayed
+        onView(withId(R.id.nav_title)).check(matches(isDisplayed()))
+
+        //Checking if My places menu item is displayed
+        onView(withText("My Places")).check(matches(isDisplayed()))
+
+        //Checking if Resources menu item is displayed
+        onView(withText("Resources")).check(matches(isDisplayed()))
+
+        //Checking if Explore menu item is displayed
+        onView(withText("Explore")).check(matches(isDisplayed()))
+
+        //Checking if Settings menu item is displayed
+        onView(withText("Settings")).check(matches(isDisplayed()))
+
+
+    }
+
+    @Test
+    fun test_isNagivationToCalendarCorrect(){
+
+        //Opening side menu by clicking on menu  button
+        onView(withId(R.id.mt_nav)).check(matches(isDisplayed())).perform(click())
+        //Checking if Schedule menu item is displayed and clicks that option
+        onView(withText("Schedule")).perform(click())
+        //Checking if that action did take you to map_fragment view
+        onView(withId(R.id.calendar_fragment)).check(matches(isDisplayed()))
+        Espresso.pressBack()
+        //Checking if that action id did take you to map_fragment view
+        onView(withId(R.id.coordinate_layout)).check(matches(isDisplayed()))
+    }
```

Comment:
I spoke with Amanda, this is indeed redundant and is no longer needed. Please remove it! Thank you and great work! 😄